source/core/shape.cc

```diff
@@ -495,6 +495,353 @@ CompositeShape<dim>::displaced_volume(const double fluid_density)
   return solid_volume;
 }
 
+
+template <int dim>
+double
+RBFShape<dim>::value(const Point<dim> &evaluation_point,
+                     const unsigned int /*component*/) const
+{
+  Point<dim> centered_point = this->align_and_center(evaluation_point);
+
+  // The computation and addition of the bounding box distance is necessary
+  // since the RBF nodes may not cover the whole simulation domain. In that
+  // case, it is assumed that the distance from the RBF object is approximately
+  // the same as the distance from the corresponding bounding box.
+  double bounding_box_distance = bounding_box->value(centered_point);
+  double value = bounding_box_distance > 0. ? bounding_box_distance : 0.;
+
+  double       dist, basis;
+  unsigned int number_of_nodes = weight.size();
+  // Algorithm inspired by Optimad Bitpit. https://github.com/optimad/bitpit
+  for (unsigned int i = 0; i < number_of_nodes; ++i)
+    {
+      dist  = (centered_point - nodes[i]).norm() / support_radius[i];
+      basis = evaluate_basis_function(basis_function[i], dist);
+      value += basis * weight[i];
+    }
+  return value;
+}
+
+template <int dim>
+std::shared_ptr<Shape<dim>>
+RBFShape<dim>::static_copy() const
+{
+  std::shared_ptr<Shape<dim>> copy =
+    std::make_shared<RBFShape<dim>>(support_radius,
+                                    basis_function,
+                                    weight,
+                                    nodes,
+                                    this->position,
+                                    this->orientation);
+  return copy;
+}
+
+template <int dim>
+double
+RBFShape<dim>::displaced_volume(const double fluid_density)
+{
+  std::cout
+    << "Warning: For a RBF shape, the real volume will be lower than output."
+    << std::endl;
```

Comment:
@paulapatience is right, but this is why we never write to std::cout. If you write to std::cout in parallel you just get a cluster fuck. Either throw if this is an error or use the parallel terminal writing capabilities.

Reply:
I tried with parallel output, but I am not convinced yet by the structure. 
I could put the warning as a comment, or throw an error since displaced_volume() is only used to integrate particle motion, and we can only do it when particles are spherical.